File: app/src/main/java/com/softaai/designpatterns/java/structural/adapter/VlcPlayer.java
package com.softaai.designpatterns.java.structural.adapter;

/**
 * Created by Amol Pawar on 02-04-2019.
 * softAai Apps
 */
public class VlcPlayer implements AdvancedMediaPlayer {
    @Override
    public void playVlc(String fileName) {
        System.out.println("Playing vlc file. Name : " + fileName);
    }

    @Override
    public void playMp4(String fileName) {
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/adapter/VlcPlayer.kt
package com.softaai.designpatterns.kotlin.structural.adapter


/**
 * Created by Amol Pawar on 02-04-2019.
 * softAai Apps
 */
class VlcPlayer : AdvancedMediaPlayer {
    override fun playVlc(fileName: String) {
       println("Playing vlc file. Name : " + fileName)
    }

    override fun playMp4(fileName: String) {
    }
}